### PR TITLE
fix promise leak

### PR DESCRIPTION
**Pull request by tanner0101**
Fixes a promise leak caused by chaining the request completion promise to the channel write.

### CODE REVIEW

File: Sources/PostgresNIO/Connection/PostgresRequest.swift [hunk 1]
```diff
@@ -5,7 +5,7 @@ extension PostgresConnection: PostgresClient {
         request.log(to: self.logger)
         let promise = self.channel.eventLoop.makePromise(of: Void.self)
         let request = PostgresRequestContext(delegate: request, promise: promise)
-        self.channel.write(request).cascadeFailure(to: promise)
```
Comment (weissi) [inline Sources/PostgresNIO/Connection/PostgresRequest.swift]: hmm, I don't think this can _fix_ a promise leak, right? If anything this will make it more likely that the promise is never fulfilled. Also, this is missing a test I think?

File: Sources/PostgresNIO/Data/PostgresData+Double.swift [hunk 2]
```diff
@@ -31,3 +31,20 @@ extension PostgresData {
         }
     }
 }
+
+extension Double: PostgresDataConvertible {
```
Comment (weissi) [inline Sources/PostgresNIO/Data/PostgresData+Double.swift]: this and the change below seem unrelated?

**Comment by tomerd** [comment]
thanks @tanner0101
